content/information-distribution/signature.md

```diff
@@ -17,28 +17,44 @@ Informo signed events are built using a common structure, inspired from [Matrix
 `m.room.encrypted`
 events](https://matrix.org/docs/spec/client_server/r0.4.0.html#m-room-encrypted):
 
-|    Field     |   Type   |                                    Description                                    |
-| ------------ | -------- | --------------------------------------------------------------------------------- |
-| `algorithm`  | `string` | Algorithm used for signing the content. 🔧 Define allowed algorithms               |
-| `sender_key` | `string` | The public part of the key used for signing the event                             |
-| `device_id`  | `string` | 🔧 Optional, ID of the sending device, may be used for Megolm                      |
-| `session_id` | `string` | 🔧 Optional, ID of the session used to encrypt the message, may be used for Megolm |
-| `signature`  | `string` | Generated signature                                                               |
-| `signed`     | `object` | Content that has been signed                                                      |
-
-<!-- 🔧: Need to do some research on Megolm and Matrix APIs around encryption and key
-management -->
+|    Field     |      Type      |     Description     |
+|--------------|----------------|---------------------|
+| `signature`  | `string`       | Generated signature |
+| `signed`     | `SignedObject` | Object to be signed |
+
+Where:
+
+* `SignedObject` is a map using the following structure:
+
+| Parameter       |       Type       | Req. |                                                  Description                                                  |
+|:----------------|:-----------------|:----:|:--------------------------------------------------------------------------------------------------------------|
+| `sender`        | `string`         |  x   | User who sent the event. **Must** match the Matrix user ID of who sent the event.                             |
+| `room_id`       | `string`         |  x   | Associated room ID. **Must** match the room ID where the event has been sent.                                 |
+| `state_key`     | `string`         |  !   | State key of the event. Required if the event is a state event.                                               |
+| `signatory`     | `string`         |      | The Matrix user ID who signed this data. If not present, defaults to the value defined in the `sender` field. |
+| `signatory_key` | `string`         |  x   | The public part of the key used for signing the event.                                                        |
+| `algorithm`     | `string`         |  x   | Algorithm used for signing the content.                                                                       |
+| `content`       | `object`         |  x   | The content of the event to be signed.                                                                        |
+
```

Comment:
Don't we also want to sign the event's type?

Reply:
yes !

Reply:
Then you should add it :)

Reply:
Nevermind, didn't notice it.